php/actions/get_torrent_files.php

```diff
@@ -106,9 +106,16 @@
             }
             $trackers = $torrent->getTrackers();
             foreach ($trackers as &$tracker) {
-                $tracker = preg_replace('/(?<==)\w+$/', $cfg['user_passkey'], $tracker);
-                if ($cfg['tor_for_user']) {
-                    $tracker = preg_replace('/\w+(?==)/', 'pk', $tracker);
+                // Если задан пустой заменный ключ, то пихаем дефолтный 'ann?magnet'
+                if (empty($cfg['user_passkey'])) {
+                    $tracker = preg_replace('/(?<=ann\?).+$/', 'magnet', $tracker);
+                } else {
+                    $tracker = preg_replace('/(?<==)\w+$/', $cfg['user_passkey'], $tracker);
+
+                    // Для обычных пользователей заменяем адрес трекера и тип ключа.
+                    if ($cfg['tor_for_user']) {
+                        $tracker = preg_replace(['/(?<=\.)([-\w]+\.\w+)/', '/\w+(?==)/'], ['t-ru.org', 'pk'], $tracker);
+                    }
```

Comment:
This will not work when the target passkey is empty.

Reply:
done